Sources/Signal.swift

```diff
@@ -27,6 +27,12 @@ public final class Signal<Value, Error: Swift.Error> {
 	/// The state of the signal. `nil` if the signal has terminated.
 	private let state: Atomic<SignalState<Value, Error>?>
 
+	/// The status of the signal. It synchronizes with `sendLock`.
+	private var status: SignalStatus<Value, Error>
```

Comment:
Nice idea. Note that I have pulled in the Read-Copy-Update change from #139 to make this work. :tada: